src/Microsoft.AspNet.Diagnostics.Elm/Views/DetailsPage.cshtml

```diff
@@ -14,56 +14,89 @@
     }
 
     public DetailsPageModel Model { get; set; }
-}
 
-@helper LogRow(LogInfo log)
-{
-    if (log.Severity >= Model.Options.MinLevel &&
-        (string.IsNullOrEmpty(Model.Options.NamePrefix) || log.Name.StartsWith(Model.Options.NamePrefix, StringComparison.Ordinal)))
+    public HelperResult LogRow(LogInfo log)
     {
-        <tr>
-            <td>@string.Format("{0:MM/dd/yy}", log.Time)</td>
-            <td>@string.Format("{0:H:mm:ss}", log.Time)</td>
-            <td class="@log.Severity.ToString().ToLowerInvariant()">@log.Severity</td>
-            <td title="@log.Name">@log.Name</td>
-            <td title="@log.Message" class="logState" width="100px">@log.Message</td>
-            <td title="@log.Exception">@log.Exception</td>
-        </tr>
-    }
-}
-
-@helper Traverse(ScopeNode node)
-{
-    var messageIndex = 0;
-    var childIndex = 0;
-    while (messageIndex < node.Messages.Count && childIndex < node.Children.Count)
-    {
-        if (node.Messages[messageIndex].Time < node.Children[childIndex].StartTime)
-        {
-            @LogRow(node.Messages[messageIndex])
-            messageIndex++;
-        }
-        else
+        return new HelperResult((__razor_helper_writer) =>
```

Comment:
Actually, now that I reflect on it we can probably get away without that HelperResult (and thus __razor_helper_writer) at all now that we have the Write and WriteLiteral overloads, which figure out what to write to for you. I'll give that a shot, and if it doesn't work I'll make up a better name.